guides/source/action_controller_overview.md

```diff
@@ -341,18 +408,50 @@ used in this way.
 params.expect(log_entry: {})
 ```
 
+The above marks the `:log_entry` parameters hash and any sub-hash of it as
+permitted and does not check for permitted scalars, anything is accepted.
+
+WARNING: Extreme care should be taken when using `permit!`, as it will allow all
+current and *future* model attributes to be mass-assigned.
+
+#### `expect`
+
+The [`expect`][] method provides a concise and safe way to require and permit parameters.
+
+```ruby
+id = params.expect(:id)
+```
+
+The above `expect` will always return a scalar value and not an array or hash.
```

Comment:
A chunk of this is duplicated from around line 391. It may also appear to be the case that an example of `permit!` is missing from above?

Reply:
Good catch! Thanks!